source/octod/media.d

```diff
@@ -0,0 +1,172 @@
+/*******************************************************************************
+
+    Wraps media type string used for interaction with GitHub as defined
+    by https://developer.github.com/v3/media
+
+    Copyright: Copyright (c) 2016 Sociomantic Labs. All rights reserved
+
+    License: Boost Software License Version 1.0 (see LICENSE for details)
+
+*******************************************************************************/
+
+module octod.media;
+
+/**
+    Stores sub-parts of media type string as separate fields. Simplifies
+    branching based on media types and safeguards against some typos in the
+    string.
+ **/
+struct MediaType
+{
+    /// Media type used by default if none is specified by user of the library
+    enum Default = MediaType.init;
+
+    immutable
+    {
+        /**
+            Indicates if this media type is of "vnd.github" ones. If set to
+            `false`, this becomes simple `application/something` media type.
+         **/
+        bool   github_marker = true;
+        /**
+            API version to be used for the request. If empty, GitHub will
+            assume whatever default API version currently is.
+         **/
+        string ver           = "v3";
+        /**
+            Custom parameters, usually indicate kind of content to serve
+            (for example, rendered markdown or raw source)
+         **/
+        string param         = "";
+        /**
+            Format of response data, usually "json" or "xml"
+         **/
+        string format        = "json";
+    }
```

Comment:
I don't mean to only allow one value but just define the existing ones as "manifest constants". (yes, it was a great idea to reuse the `enum` keyword for manifest constants :trollface: )